### PR TITLE
[#218] Adds linky filter to rendered message body to render clickable links

### PR DESCRIPTION
**Pull request by pamo**
Angular supports #218 out of the box with the [`linky`](https://docs.angularjs.org/api/ngSanitize/filter/linky) filter. Had to do a lot of CSS refactoring as a part of this to avoid the rendered `<a>` tags in the card body from being styled with the styles we were applying to message card action buttons.

linky

### CODE REVIEW

File: components/mainContent.html [hunk 1]
```diff
@@ -22,17 +22,22 @@ <h2 ng-show="!editing" ng-click="editing = true; newName=type.value;">
               <input id="new_name_{{ type.id }}" ng-model="newName" focus></input>
               <div>
                 <button ng-click="changeColumnName(type.id, newName);editing=false" class="success-button">Save</button>
-                <a class="blue" ng-click="editing=false;">Cancel</a>
+                <button class="blue"
+                        type="button"
```
Comment (glauberramos) [inline components/mainContent.html]: does it need type="button" if it is already a button?

File: components/mainContent.html [hunk 2]
```diff
@@ -50,16 +55,23 @@ <h2 ng-show="!editing" ng-click="editing = true; newName=type.value;">
...
-                      </div>
+                    <div class="editing">
+                      <textarea data-emoji-picker="true"
+                                ng-model="message.text"
+                                ng-change="messages.$save(message)"
+                                aria-label="Message text entry"></textarea>
+
+                      <button ng-click="isEditing = false;saveMessage(message); updateEditingMessage(message, false);"
+                              type="button"
+                              class="message-action -done">DONE</button>
```
Comment (glauberramos) [inline components/mainContent.html]: -done don't make sense as a name. it should be message-action-done

File: components/mainContent.html [hunk 3]
```diff
@@ -50,16 +55,23 @@ <h2 ng-show="!editing" ng-click="editing = true; newName=type.value;">
...
+                      <button class="message-action -delete"
```
Comment (glauberramos) [inline components/mainContent.html]: same here

File: stylesheets/_button.scss [hunk 4]
```diff
@@ -63,25 +63,50 @@ header button:hover {
...
-  background: $green;
+  font-size: 0.7em;
+
+  &.-delete,
+  &.-edit {
+    right: 10px;
+    font-size: 1em;
+  }
+
+  &.-delete {
```
Comment (glauberramos) [inline stylesheets/_button.scss]: remove the .

File: stylesheets/_button.scss [hunk 5]
```diff
@@ -63,25 +63,50 @@ header button:hover {
...
+  border: 0;
+  background: transparent;
   padding: 0;
-  margin-left: 0;
-  font-size: 0.7em;
   line-height: 2em;
-  background: $green;
+  font-size: 0.7em;
+
+  &.-delete,
```
Comment (glauberramos) [inline stylesheets/_button.scss]: remove .

File: stylesheets/_button.scss [hunk 6]
```diff
@@ -63,25 +63,50 @@ header button:hover {
...
+  &.-edit {
```
Comment (glauberramos) [inline stylesheets/_button.scss]: remove .

File: stylesheets/_button.scss [hunk 7]
```diff
@@ -63,25 +63,50 @@ header button:hover {
...
+  &.-delete {
+    bottom: 10px;
+    color: $danger;
+
+    &:hover {
+      color: darken($danger, 15%);
+    }
+  }
+
+  &.-edit {
```
Comment (glauberramos) [inline stylesheets/_button.scss]: same

File: stylesheets/_button.scss [hunk 8]
```diff
@@ -63,25 +63,50 @@ header button:hover {
...
+    top: 10px;
+  }
+
+  &.-done {
```
Comment (glauberramos) [inline stylesheets/_button.scss]: same

File: stylesheets/_button.scss [hunk 9]
```diff
@@ -63,25 +63,50 @@ header button:hover {
...
+    &:hover {
+      bottom: 13px;
+      border-bottom: 0;
+    }
+  }
 }
 
 .success-button,
-.message button {
+.message-action.-done {
```
Comment (glauberramos) [inline stylesheets/_button.scss]: same

File: stylesheets/_button.scss (hunk 4 above)
Comment (pamo) [reply stylesheets/_button.scss]: Redefining it this way will require applying both `message-action` and `message-action-done` in the template class attribute. Is that ok with you?

```
class="message-action message-action-done"
```

File: components/mainContent.html (hunk 1 above)
Comment (pamo) [reply components/mainContent.html]: This is habitual for me. I've seen some cases where a button without a type will be interpreted as a form submit button by default. Explicitly declaring the button type indicates client side behavior. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#type

File: stylesheets/_button.scss (hunk 4 above)
Comment (glauberramos) [reply stylesheets/_button.scss]: yeah that's ok to have both classes

File: components/mainContent.html (hunk 1 above)
Comment (glauberramos) [reply components/mainContent.html]: ok

File: stylesheets/_button.scss (hunk 4 above)
Comment (pamo) [reply stylesheets/_button.scss]: Cool I'll make the change

**Comment by pamo** [comment]
Made the requested changes and fixed a bug I introduced with the enterClick directive @glauberramos

**Comment by glauberramos** [comment]
There is 2 small things that we need to fix:
- Style the links with an underline
- The behavior before was that when I type a letter, the button "done" would move the the next line, now the text is behind the button. Maybe that is happening because you changed from a link to a button, but I am not sure.

screen shot 2017-10-28 at 19 14 26
screen shot 2017-10-28 at 19 14 34

**Comment by pamo** [comment]
Whoa good finds, I'll update the PR soon.

**Comment by pamo** [comment]
Made updates to the PR, @glauberramos. Can you take another look?

**Comment by glauberramos** [comment]
Hey @pamo,

Sorry but I can't accept this PR since it changed too much the design. I would prefer only the Link functionality.

Right now the buttons have changed place and the cards look bigger.

I will add the linky functionality and close this.

**Comment by pamo** [comment]
@Glauberamos kk. you'll see that the specificity of the defined styles affect the way the links are rendered once you add the linky filter. I guess my clean-up went a little out of hand.

**Pull request closed without merging**